client-common/src/key/private_key.rs

```diff
@@ -6,10 +6,81 @@ use zeroize::Zeroize;
 
 use crate::{ErrorKind, PublicKey, Result, ResultExt, SECP};
 
+/// types of private key
+#[derive(Clone)]
+pub enum PrivateKeyKind {
+    /// normal private key
+    Normal(PrivateKey),
+    // TODO: Add Hardware wallet as Privatekey
+}
```

Comment:
maybe:
```
/// temporary place holder
pub type HWalletConfig = ();
/// location
pub enum PrivateKeyLocation {
/// private key in local filesystem
Local(PrivateKey),
External(HWalletConfig),
}

Reply:
in match statements, it could do `todo!()` for the ops at the moment.

one can still test the key management -- e.g. if one can create a wallet with these external keys (and say display / return this information)

Reply:
yes, just add dummy enum
HW_LEDGER(ledger),
HW_TREASURE(treasure)